pkg/services/k8s/resources/publicdashboard/webhooks.go

```diff
@@ -152,3 +173,112 @@ func (api *WebhooksAPI) Create(c *contextmodel.ReqContext) response.Response {
 
 	return response.JSON(int(resp.Response.Result.Code), resp)
 }
+
+func (api *WebhooksAPI) MutationCreate(c *contextmodel.ReqContext) response.Response {
+	api.Log.Debug("mutation controller create fired")
+	body, err := io.ReadAll(c.Req.Body)
+	if err != nil {
+		api.Log.Error("error reading request body")
+	}
+	api.Log.Debug("create", "body", string(body))
+
+	var rev k8sAdmission.AdmissionReview
+	err = json.Unmarshal(body, &rev)
+	if err != nil {
+		api.Log.Error("error unmarshalling request body")
+		return response.Error(500, "error unmarshalling request body", err)
+	}
+
+	// Current state of object
+	obj := &PublicDashboard{}
+	err = obj.UnmarshalJSON(rev.Request.Object.Raw)
+	if err != nil {
+		api.Log.Error("error unmarshalling request body")
+		return response.Error(500, "error unmarshalling request body", err)
+	}
+
+	// Previous state of object for update or delete
+	oldObj := &PublicDashboard{}
+	err = oldObj.UnmarshalJSON(rev.Request.OldObject.Raw)
+	if err != nil {
+		api.Log.Error("error unmarshalling request body")
+		return response.Error(500, "error unmarshalling request body", err)
+	}
+
+	// THIS IS BROKEN
+	// TODO: convert error to k8sAdmission.AdmissionResponse and then to response.Response
+	req := &admission.AdmissionRequest{
+		Action:  c.Req.Method,
+		Kind:    rev.Kind,
+		Group:   rev.GroupVersionKind().Group,
+		Version: rev.GroupVersionKind().Version,
+		UserInfo: admission.AdmissionUserInfo{
+			Username: rev.Request.UserInfo.Username,
+			UID:      rev.Request.UserInfo.UID,
+			Groups:   rev.Request.UserInfo.Groups,
+		},
+		Object:    obj,
+		OldObject: oldObj,
+	}
+
+	var resp *k8sAdmission.AdmissionReview
+	mutationResponse, err := api.MutationController.Mutate(c.Req.Context(), req)
+	if err != nil {
+		// auth status code to start, maybe change this to bad request
+		resp = makeFailureAdmissionReview(rev.Request.UID, rev.TypeMeta, err, 403)
+	} else {
+		resp = makeSuccessfulAdmissionReview(rev.Request.UID, rev.TypeMeta, 200)
+	}
+
+	// compare original and defaulted version
```

Comment:
@jalevin looks like we still have a couple more TOOOs on this:

- [ ] the `runtime.Object` needs to be marshaled to json `[]byte`
- [ ] `jsonpatch.CreatePatch` using `rev.Request.Object.Raw` and the bytes from the step above
- [ ] add that to the `review.Response.Patch`
- [ ] check into how errors are handled for mutation hooks